tests/test_timeseries.py

```diff
@@ -0,0 +1,274 @@
+"""Test timeseries module"""
+
+from datetime import datetime
+from typing import Dict, Tuple
+
+import pytest
+from dateutil.relativedelta import relativedelta
+from fastapi import HTTPException
+
+# Import your functions here
+from titiler.cmr.timeseries import (
+    TimeseriesParams,
+    generate_datetime_ranges,
+    parse_duration,
+    timeseries_query,
+)
+
+
+def test_parse_duration():
+    """Test durations"""
+    assert parse_duration("P1Y") == relativedelta(years=1)
+    assert parse_duration("P2M") == relativedelta(months=2)
+    assert parse_duration("P3D") == relativedelta(days=3)
+    assert parse_duration("PT4H") == relativedelta(hours=4)
+    assert parse_duration("PT5M") == relativedelta(minutes=5)
+    assert parse_duration("PT6S") == relativedelta(seconds=6)
+    assert parse_duration("PT1S") == relativedelta(seconds=1)
+    assert parse_duration("P1Y2M3DT4H5M6S") == relativedelta(
+        years=1, months=2, days=3, hours=4, minutes=5, seconds=6
+    )
```

Comment:
you can use `pytest.mark.parametrize` 

https://github.com/cogeotiff/rio-tiler/blob/9a39c802044b485a26c29065b539ec44c817d8a6/tests/test_expression.py#L15-L27

Reply:
I wound up finding a lightweight import (`isodate`) that has a `parse_duration` function that we can use instead of a home cooked regex function.